Sources/BSON/BSONDecimal128.swift

```diff
@@ -0,0 +1,505 @@
+             * In order to preserve compatibility with the densely packed decimal
+             * format, the maximum value for the significand of decimal128 is
+             * 1e34 - 1.  If the value is greater than 1e34 - 1, the IEEE 754
+             * standard dictates that the significand is interpreted as zero.
+             */
+            isZero = true
+        }
+
+        if isZero {
+            significandDigits = ["0"]
+        } else {
+            for _ in 0...3 {
+                var (quotient, remainder) = significand128.divideBy1Billion()
+                significand128 = quotient
+                // We now have the 9 least significant digits.
+                for _ in 0...8 {
+                    significandDigits.insert(Character(String(remainder % 10, radix: 10)), at: 0)
+                    remainder /= 10
+                }
+            }
+        }
+
+        if !isZero, let firstNonZero = significandDigits.firstIndex(where: { $0 != "0" }) {
+            significandDigits = [Character](significandDigits.suffix(from: firstNonZero))
+        }
+
+        // Exponential - [-]d.ddde(+/-)dd or [-]de(+/-)dd
+        // Regular     - ddd.ddd
+
+        /*
+         * The adjusted_exponent checks are dictated by the string conversion
+         * specification.
+         *
+         * We must check exponent > 0, because if this is the case, the number
+         * has trailing zeros.  However, we *cannot* output these trailing zeros,
+         * because doing so would change the precision of the value, and would
+         * change stored data if the string converted number is round tripped.
+         */
+        var representation = self.isNegative ? "-" : ""
+
+        let adjusted_exponent = exponent + (significandDigits.count - 1)
+        if exponent > 0 || adjusted_exponent < -6 {
+            // Exponential format
+            representation += String(significandDigits[0])
+            representation += significandDigits.count > 1 ? "." : ""
+            representation += String(significandDigits[1..<significandDigits.count])
+            representation += "E"
+            representation += String(format: "%+d", adjusted_exponent)
+        } else {
+            // Regular format
+            guard exponent != 0 else {
+                representation += String(significandDigits)
+                return representation
+            }
+
+            var pointPosition = significandDigits.count + exponent
+
+            if pointPosition > 0 {
+                // number isn't a fraction
+                representation += String(significandDigits[0..<pointPosition])
+                significandDigits = Array(significandDigits[pointPosition...])
+            } else {
+                representation += "0"
+            }
+
+            representation += "."
+
+            if pointPosition < 0 {
+                representation += String(repeating: "0", count: abs(pointPosition))
+                pointPosition += abs(pointPosition)
```

Comment:
I don't think `pointPosition` is used after this, so we don't need to modify it. As part of that, we can make it a `let` constant above.